### PR TITLE
[Chore] tools.images_packer: refine previous offline_ota_image_builder to support 2 image packing mode

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Description

This PR refines the `tools.offline_ota_image_builder`(renamed to `images_packer`) to explicitly provides two image packing mode(each requires different command args), which:

1. `build-cache-src`: build external cache source image and (optionally) prepare specific device as external cache source device,

2. `build-offline-ota-imgs-bundle`: build offline OTA image bundle for offline OTA.

> [!NOTE]
> fully offline OTA helper script is not yet implemented.

For detailed usage, please check `tools/images_packer/README.md`.

### CODE REVIEW

**Review by obi-t4 — COMMENTED** [review]
I left some comments, please have a check!

File: tools/images_packer/__main__.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        _parsed = str(raw_pair).split(":")
+        if len(_parsed) >= 2:
+            _ecu_id, _image_fpath, _image_version, *_ = *_parsed, ""
+            image_metas.append(
+                ImageMetadata(
+                    ecu_id=_ecu_id,
+                    image_version=_image_version,
+                )
+            )
+            image_files[_ecu_id] = _image_fpath
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply tools/images_packer/__main__.py:79]: Updated at [a9edc6b](https://github.com/tier4/ota-client/pull/267/commits/a9edc6bcabd49d881e17700e266eb214f03b789f).

[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply tools/images_packer/__main__.py:79]: https://github.com/tier4/ota-client/blob/a9edc6bcabd49d881e17700e266eb214f03b789f/tools/images_packer/__main__.py#L72-L79

**Comment by obi-t4** [comment]
@Bodong-Yang 
bash script has been added at:
https://github.com/tier4/ota-client/pull/267/commits/707857c889682e8d5faaf3f63deba242e798fca5

File: tools/images_packer/__main__.py [hunk 2]
```diff
@@ -69,6 +69,13 @@ def main_build_offline_ota_image_bundle(args: argparse.Namespace):
                     image_version=_image_version,
                 )
             )
+            if _ecu_id in image_files:
```
Comment (obi-t4) [inline tools/images_packer/__main__.py:72]: I think this should not be warning. For the user, this is the unexpected behavior, so it should be error.

Comment (Bodong-Yang) [reply tools/images_packer/__main__.py:72]: Yes I agree! This is unexpected behavior. Let me fix it.

Comment (Bodong-Yang) [reply tools/images_packer/__main__.py:72]: Updated at [786ef4d](https://github.com/tier4/ota-client/pull/267/commits/786ef4d160af649bf21ca82f42b5e30b34eca644), now multiple images specified to the same ECU will cause the images_packer logging an error and then exit.

File: tools/images_packer/__main__.py [hunk 3]
```diff
@@ -70,12 +70,14 @@ def main_build_offline_ota_image_bundle(args: argparse.Namespace):
...
-                logger.warning(
+                logger.error(
                     (
-                        f"override previously set OTA target image for ECU@{_ecu_id} "
-                        f"from {image_files[_ecu_id]} to {_image_fpath}"
+                        f"ERR: OTA target image for ECU@{_ecu_id} is already set to"
+                        f"{image_files[_ecu_id]}, please check the --image params"
                     )
                 )
+                sys.exit(errno.EINVAL)
```
Comment (obi-t4) [inline tools/images_packer/__main__.py:79]: :+1:

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Pull request merged**